### PR TITLE
Import configuration files instead of opening them with notepad

### PR DESCRIPTION
**Pull request by ValdikSS**
https://github.com/OpenVPN/openvpn-gui/pull/76

### CODE REVIEW

**Comment by mattock** [comment]
Quoting the previous community [meeting summary](https://sourceforge.net/p/openvpn/mailman/message/35295113/):

    Discussed an issue with .ovpn file associations in Windows. It would be 
    best if clicking on a .ovpn file would bring up a gui that asks "do you 
    want to import this, or start it right away".

Does the user get any clue as to what is happening when he/she clicks on an .ovpn file when this patch is included?

**Comment by ValdikSS** [comment]
@mattock yes, the user is prompted with a dialog "Do you want to import file blabla.ovpn?" "Yes/No". This commit also introduces an "edit" item in right-click submenu for .ovpn files which opens notepad.exe.

**Comment by ValdikSS** [comment]
@mattock and of course there is/was "Start OpenVPN on this config file" right-click menu item.

File: windows-nsis/openvpn.nsi [hunk 1]
```diff
@@ -336,9 +336,11 @@ Section /o "${PACKAGE_NAME} File Associations" SecFileAssociation
 	WriteRegStr HKCR "${PACKAGE_NAME}File" "" "${PACKAGE_NAME} Config File"
 	WriteRegStr HKCR "${PACKAGE_NAME}File\shell" "" "open"
 	WriteRegStr HKCR "${PACKAGE_NAME}File\DefaultIcon" "" "$INSTDIR\icon.ico,0"
-	WriteRegStr HKCR "${PACKAGE_NAME}File\shell\open\command" "" 'notepad.exe "%1"'
+	WriteRegStr HKCR "${PACKAGE_NAME}File\shell\open\command" "" '"$INSTDIR\bin\openvpn-gui.exe" --import "%1"'
 	WriteRegStr HKCR "${PACKAGE_NAME}File\shell\run" "" "Start ${PACKAGE_NAME} on this config file"
 	WriteRegStr HKCR "${PACKAGE_NAME}File\shell\run\command" "" '"$INSTDIR\bin\openvpn.exe" --pause-exit --config "%1"'
+	WriteRegStr HKCR "${PACKAGE_NAME}File\shell\edit" "" "Edit config file"
+	WriteRegStr HKCR "${PACKAGE_NAME}File\shell\edit\command" "" 'notepad.exe "%1"'
```
Comment (mattock) [inline windows-nsis/openvpn.nsi]: As mentioned in [https://github.com/OpenVPN/openvpn-gui/pull/76](openvpn-gui PR 76) these two lines should be removed and the relevant code in openvpn-gui made to use file associations instead. Besides that this looks good. I also tested this patch in isolation and it worked as intended, trying to launch openvpn-gui.exe --import.

**Comment by jktech89** [comment]
From windows client trying to use --import file.ovpn returns unknown error even when in the same program files directory or config folder it doesn't pick it up w/o manually importing via menu.  Ideas?

**Comment by selvanair** [comment]
> From windows client trying to use --import file.ovpn returns unknown error even when in the same program files directory or config folder it doesn't pick it up w/o manually importing via menu. Ideas?

Are you referring to this PR or the released version?  Currently there is no --import option. But if you manually copy the config to the global or local config folder (or sub folders) it will automatically appear in the GUI menu. Same as importing from the menu. If that's not working please open an issue in the openvpn-gui repo.

**Comment by flichtenheld** [comment]
We have decided to remove the nsis/mingw buildsystem since it is not maintained anymore. Only the msi/msvc buildsystem will remain, unless someone steps up to maintain the other parts. This issue will be closed when the removal actually happens.

**Comment by selvanair** [comment]
This PR is obsolete as the corresponding PR in the GUI has been superseded by recent changes in import functionality, import from URL etc. It may be closed.

**Pull request closed without merging**